### DISCUSSION TITLE
How to implement a chess variant where you choose a restriction for your opponent's move as part of your move.

### DISCUSSION THREAD

**Discussion by gerzytet** [fairy-stockfish/Fairy-Stockfish#557] [Q&A]
How do I implement a chess variant where you choose a restriction for your opponent's move as part of your move?

Specifically, the one I want to implement is choosing a piece type that the opponent isn't allowed to move.  How do I do this?

**Comment by ianfab** [comment]
One of the major aspects of this is that it needs to be encoded in the move representation and accordingly be considered in move generation and validation.

In a way this is pretty similar to what I did for duck chess, so you might want to have a look at that to get a feeling for what needs to be changed.
